File: iOS/Swift/Xevensthein/ViewController.swift
//
//  ViewController.swift
//  Xevensthein
//
//  Created by Antonio Feregrino Bolaños on 4/29/16.
//  Copyright © 2016 That C# guy. Some rights reserved.
//

import UIKit
import XevenstheinCore

class ViewController: UIViewController, UITextFieldDelegate {
    
    // MARK: Properties
    @IBOutlet var firstWordTextField: UITextField!
    @IBOutlet var secondWordTextField: UITextField!
    @IBOutlet var compareButton: UIButton!
    @IBOutlet var resultLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        firstWordTextField.delegate = self;
        secondWordTextField.delegate = self;
        
        compareButton.addTarget(self, action: #selector(ViewController.compareButtonClicked(_:)), forControlEvents: .TouchUpInside);
    }
    
    func compareButtonClicked(_sender : AnyObject?){
        let result = LevenshteinDistance.compute( firstWordTextField.text!,t: secondWordTextField.text!);
        resultLabel.text = String(result);
    }
    
    func textFieldShouldReturn(textField: UITextField) -> Bool {
        if(textField == firstWordTextField){
            firstWordTextField.resignFirstResponder();
            secondWordTextField.becomeFirstResponder();
        }
        else{
            secondWordTextField.resignFirstResponder();
        }
        return true;
    }
    

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

File: iOS/C#/ViewController.cs
using System;
using UIKit;
using Xevensthein.Core;

namespace Xevensthein
{
    public partial class ViewController : UIViewController, IUITextFieldDelegate
    {
        public ViewController(IntPtr handle)
            : base(handle)
        {
           
        }

        public override void ViewDidLoad()
        {
            base.ViewDidLoad();

            FirstWordTextField.Delegate = this; 
            SecondWordTextField.Delegate = this;

            CompareButton.TouchUpInside += CompareButtonClicked;
        }

        void CompareButtonClicked (object sender, EventArgs e)
        {
            var result = LevenshteinDistance.Compute(FirstWordTextField.Text, SecondWordTextField.Text);
            ResultLabel.Text = result.ToString();
        }
            
        [Foundation.Export("textFieldShouldReturn:")]
        public bool ShouldReturn(UIKit.UITextField textField)
        {
            if (textField == FirstWordTextField)
            {
                FirstWordTextField.ResignFirstResponder();
                SecondWordTextField.BecomeFirstResponder();
            }
            else
            {
                SecondWordTextField.ResignFirstResponder();
            }
            return true;
        }

        public override void DidReceiveMemoryWarning()
        {
            base.DidReceiveMemoryWarning();
            // Release any cached data, images, etc that aren't in use.
        }
    }
}

